mmdet/core/bbox/samplers/random_sampler.py

```diff
@@ -51,7 +51,10 @@ def random_choice(self, gallery, num):
             else:
                 device = 'cpu'
             gallery = torch.tensor(gallery, dtype=torch.long, device=device)
-        perm = torch.randperm(gallery.numel(), device=gallery.device)[:num]
+        # This is a temporary fix. We can revert the following code
+        # when PyTorch fixes the abnormal return of torch.randperm.
+        # See: https://github.com/open-mmlab/mmdetection/pull/5014
+        perm = torch.randperm(gallery.numel())[:num].to(device=gallery.device)
```

Comment:
I think we should keep the original logic that first perform randperm then pick the expected number of items. It may make the randperm meaningless if we always select the expected number of items then randperm

Reply:
Original code:

```py
perm = torch.randperm(gallery.numel(), device=gallery.device)[:num]
```

New code:

```py
perm = torch.randperm(gallery.numel())[:num].to(device=gallery.device)
```

If we ignore the device part:

Original code:

```py
perm = torch.randperm(gallery.numel())[:num]
```

New code:

```py
perm = torch.randperm(gallery.numel())[:num]
```

It seems that the logic has not changed.

Reply:
Ops, my mistake.